test/powershell/Language/Scripting/DeserializedTypeConversion.Tests.ps1

```diff
@@ -0,0 +1,283 @@
+﻿Describe "Tests conversion of deserialized types to original type using object properties." -Tags "CI" {
+    BeforeAll {    
+        # Create new types and test functions.
+        $type1 = Add-Type -PassThru -TypeDefinition @'
```

Comment:
I think this could be one call to Add-Type like this:

```powershell
$type1,$type2,$type3,$type4 = Add-Type -PassThru -TypeDefinition @'
...
'@
```